### ISSUE TITLE
Atom WriteToBuffer error spam when running rhi=null dedicated server inside the Editor with ctrl-g using o3de-multiplayersample samplebase level

### ISSUE THREAD

**Issue by AMZN-Olex** [o3de/o3de#8283]
**Describe the bug**
When running multiplayer server inside the Editor with CTRL+G, the following gets spam on the server and is sent over to the Editor console.

```
(EditorServer) - ==================================================================
(EditorServer) - AZ::Render::WriteToBuffer: Trace::Error
(EditorServer) -  C:/git/o3de/Gems/Atom/Utils/Code/Source/Utils.cpp(43): 'class AZ::Outcome<void,void> __cdecl AZ::Render::WriteToBuffer(class AZStd::intrusive_ptr<class AZ::RHI::Buffer>,const void *,unsigned __int64)'
(EditorServer) - AZ::Render::WriteToBuffer: Unable to map buffer: BoneTransformBuffer_burtactor, map request failed.
(EditorServer) - AZ::Render::WriteToBuffer: ==================================================================
```

**Steps to reproduce**
1. Set up o3de-multiplayersample using https://github.com/o3de/o3de-multiplayersample/blob/development/README.md
2. Launch the Editor
3. Open SampleBase level
4. set editorsv_rhi_override=null
5. ctrl+g to enter multiplayer
6. wait for the server to launch and the editor to connect as a client
7. observe the Editor console

**Expected behavior**
No error spam.

**Actual behavior**
The following error gets spammed a lot.

```
(EditorServer) - ==================================================================
(EditorServer) - AZ::Render::WriteToBuffer: Trace::Error
(EditorServer) -  C:/git/o3de/Gems/Atom/Utils/Code/Source/Utils.cpp(43): 'class AZ::Outcome<void,void> __cdecl AZ::Render::WriteToBuffer(class AZStd::intrusive_ptr<class AZ::RHI::Buffer>,const void *,unsigned __int64)'
(EditorServer) - AZ::Render::WriteToBuffer: Unable to map buffer: BoneTransformBuffer_burtactor, map request failed.
(EditorServer) - AZ::Render::WriteToBuffer: ==================================================================
(EditorServer) - AZ::Render::WriteToBuffer:
(EditorServer) - ==================================================================
(EditorServer) - AZ::Render::WriteToBuffer: Trace::Error
(EditorServer) -  C:/git/o3de/Gems/Atom/Utils/Code/Source/Utils.cpp(43): 'class AZ::Outcome<void,void> __cdecl AZ::Render::WriteToBuffer(class AZStd::intrusive_ptr<class AZ::RHI::Buffer>,const void *,unsigned __int64)'
(EditorServer) - AZ::Render::WriteToBuffer: Unable to map buffer: BoneTransformBuffer_burtactor, map request failed.
(EditorServer) - AZ::Render::WriteToBuffer: ==================================================================
...
```

**Assets required**
o3de-multiplayersample

**Screenshots/Video**
![image](https://user-images.githubusercontent.com/5432499/157975195-7be6f891-9765-4998-976b-dfe03586b00e.png)

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU Ryzen 5900x
 - GPU NVidia RTX 3090
 - Memory 64GB

**Comment by amzn-tommy** [comment]
I have a fix for this up for review https://github.com/o3de/o3de/pull/8472

**Comment by amzn-tommy** [comment]
@AMZN-Olex this should be fixed with commit 0c432e4b97067bd284b13b63c3c9a1a5fc82612f
